core/src/main/resources/error/error-classes.json

```diff
@@ -3,6 +3,34 @@
     "message" : [ "Field name %s is ambiguous and has %s matching fields in the struct." ],
     "sqlState" : "42000"
   },
+  "CANNOT_CAST_DATATYPE" : {
+    "message" : [ "Cannot cast %s to %s." ],
+    "sqlState" : "22018"
+  },
+  "CANNOT_CHANGE_DECIMAL_PRECISION" : {
+    "message" : [ "%s cannot be represented as Decimal(%s, %s)." ],
+    "sqlState" : "HY104"
```

Comment:
I believe that HY represents CLI-specific conditions. This seems to be triggered by overflows; maybe 22023 is a better fit.

Reply:
Yeah. Maybe 22023 is a better fit. The exception is triggered by invalid parameter value.